lib/resque_message_handler.rb

```diff
@@ -109,6 +118,17 @@ def update
 
   private
 
+  # Given a hash relating barcodes to scsb items, returns a new hash with two 
+  # keys :available & :unavailable, each of which is a hash relating barcodes
+  # to scsb documents that are Available and Not Available respectively.
+  def segment_by_availability(barcode_mapping)
+    barcode_mapping.inject({ available: {}, unavailable: {}}) do |h, (barcode, item)|
+      group = if item['availability'] == 'Available' then :available else :unavailable end
+      h[group][barcode] = item
+      h
+    end
+  end
+
```

Comment:
1) I think you should probably use the hash.select method to get the the available/unavailable items
2) Barring that, I would suggest changing the if then else construction to a ternary and changing inject to reduce

Reply:
I can use select and refactor to have function return only the segment of the hash that matches. Very curious to understand how you'd use `reduce` here though as I've always - perhaps naively - used `inject` to do reduce-style things in ruby

Reply:
I feel I can second this suggestion. I am a bit confused (but then again I am still learning Ruby).